ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRao.java

```diff
@@ -46,9 +53,37 @@ public String getVersion() {
         }
         crac.generateValidityReport(network);
 
+        // compute maximum loop flow value F_(0,all)_MAX, and update it for each Cnec in Crac
+        LoopFlowExtensionParameters loopFlowExtensionParameters = parameters.getExtension(LoopFlowExtensionParameters.class);
+        CracLoopFlowExtension cracLoopFlowExtension = crac.getExtension(CracLoopFlowExtension.class);
+        if (!Objects.isNull(loopFlowExtensionParameters) && useLoopFlowExtension(loopFlowExtensionParameters)
+            && !Objects.isNull(cracLoopFlowExtension)) {
+            //For the initial Network, compute the F_(0,all)_init
+            LoopFlowComputation loopFlowComputation = new LoopFlowComputation(network, crac, cracLoopFlowExtension.getGlskProvider(), cracLoopFlowExtension.getCountriesForLoopFlow());
+            Map<String, Double> fZeroAll = loopFlowComputation.calculateLoopFlows();
+            updateCnecsLoopFlowConstraint(crac, fZeroAll);
+        }
+
         // run optimisation
         RaoComputationResult result = Tree.search(network, crac, variantId, parameters).join();
         return CompletableFuture.completedFuture(result);
     }
 
+    public void updateCnecsLoopFlowConstraint(Crac crac, Map<String, Double> fZeroAll) {
+        // For each Cnec, get the maximum F_(0,all)_MAX = Math.max(F_(0,all)_init, loop flow threshold
+        crac.getCnecs().forEach(cnec -> {
+            CnecLoopFlowExtension cnecLoopFlowExtension = cnec.getExtension(CnecLoopFlowExtension.class);
+            if (!Objects.isNull(cnecLoopFlowExtension)) {
+                //!!! note here we use the result of branch flow of preventive state for all cnec of all states
+                //this could be ameliorated by re-calculating loopflow for each cnec in curative state: [network + cnec's contingencies + current applied remedial actions]
+                double initialLoopFlow = fZeroAll.get(cnec.getNetworkElement().getId());
+                double inputLoopFlow = cnecLoopFlowExtension.getInputLoopFlow();
```

Comment:
I would rename this to "loopFlowThreshold"

Reply:
OK